### ISSUE TITLE
'x = null' initializer in constructor does not overwrite value from field initializer

### ISSUE THREAD

**Issue by alexmarkov** [dart-lang/sdk#34499]
```dart
class A {
  int x = 5;
  A() : x = null;
}

void main() {
  print (new A().x);
}
```

```
dart foo.dart
5
dart --no-preview-dart-2 foo.dart
null
```

If field is initialized with a non-null value in a constructor, then it works fine:

```dart
class A {
  int x = 5;
  A() : x = 6;
}

void main() {
  print (new A().x);
}
```

```
dart foo.dart
6
dart --no-preview-dart-2 foo.dart
6
```

**Comment by lrhn** [comment]
Dart2js does the right thing, so it's probably a VM proper error, and not CFE related.

**Comment by srawlins** [comment]
This was fixed at some point; dart 2.6.0-dev.3.0 properly prints `null`.

**Comment by alexmarkov** [comment]
Most probably this is fixed by 46cef9bfddf3cfc05618448d228f0e3377058baf.
